### DISCUSSION TITLE
Get []u8 from IP addr struct

### DISCUSSION THREAD

**Discussion by MikhailMS** [vlang/v#19677] [Questions and Answers]
Hello there,

I am currently trying to implement a library (to work with RADIUS protocol) in V and as part of the work I need to be able to translate IPv4/IPv6 strings into bytes

I can see that there is a function `resolve_ipaddrs` in `std.net` which almost does the job required - as in it parses and creates IP struct, however there is no way to get underlying bytes of those IP structs as they are private and there are no getters

Is there currently no way to do so or am I missing something (which is possible as I am just starting with V)?

Essentially, what would be great is to be able to (or maybe even to have a stand-alone function to do the parsing)
```v
import net

resolved_ip := net.resolve_ipaddrs("192.1.10.1", net.AddrFamily.ip, net.SocketType.udp)!
ip_bytes := resolved_ip[0].addr.Ip.addr // so that ip_bytes: [4]u8 { 192, 1, 10, 1 }

resolved_ip6 := net.resolve_ipaddrs("fc66::1", net.AddrFamily.ip6, net.SocketType.udp)!
ip6_bytes := resolved_ip6[0].addr.Ip6.addr // so that ip_bytes: [16]u8 { 252, 102, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 }
```

**Comment by JalonSolov** [comment]
If you get it as a string now, you can do `str.bytes()` to get the u8 values.
```v
str := 'hello'

println(str.bytes())
println(typeof(str.bytes()).name)
```
output is
```
[104, 101, 108, 108, 111]
[]u8
```

**Reply by MikhailMS** [reply]
But that's slightly different - those are chars translated to bytes, not an ip address translated into bytes

So for inputs from the example `str.bytes()` would give wrong output

**Comment by gimpel-cs** [comment]
For IPv4 addresses the function `addr2bytes` may be a solution to your problem.

```
import arrays

fn addr2bytes(addr string) [4]u8 {
	mut bytes := [4]u8{}
	for i, x in addr.split('.') {
		bytes[i] = x.u8() 
	}
	return bytes
}

fn main() {
	addr := '192.1.1.12'
	bytes := addr2bytes(addr)
	println("addr = ${addr}")
	println("bytes = ${bytes}")
	println("type(bytes) = ${typeof(bytes).name}")
}
```
Output:
```
addr = 192.1.1.12
bytes = [192, 1, 1, 12]
type(bytes) = [4]u8
```
I've  developed this program using the V Playground with v 0.4.1. Meanwhile there may be a more elegant solution using `arrays.each`. I'm a v newbie and have problems to get this run.
